src/storage/hummock_sdk/src/slice_transform.rs

```diff
@@ -215,10 +219,94 @@ impl SliceTransform for MultiSliceTransform {
     }
 }
 
+#[derive(Default)]
+pub struct SliceTransformManagerInner {
+    table_id_to_slice_transform: RwLock<HashMap<u32, SliceTransformImpl>>,
+    version: AtomicU64,
+    notify: Notify,
+}
+
+impl SliceTransformManagerInner {
+    fn update(&self, table_id: u32, slice_transform: SliceTransformImpl) {
+        self.table_id_to_slice_transform
+            .write()
+            .insert(table_id, slice_transform);
+    }
+
+    fn remove(&self, table_id: u32) {
+        self.table_id_to_slice_transform.write().remove(&table_id);
+    }
+
+    async fn acquire(&self, mut remaining_table_id_set: HashSet<u32>) -> MultiSliceTransform {
+        let mut multi_slice_transform = MultiSliceTransform::default();
+
+        while !remaining_table_id_set.is_empty() {
+            {
+                let guard = self.table_id_to_slice_transform.read();
+                remaining_table_id_set.drain_filter(|table_id| match guard.get(table_id) {
+                    Some(slice_transform) => {
+                        multi_slice_transform.register(*table_id, slice_transform.clone());
+                        true
+                    }
+
+                    None => false,
+                });
+            }
+
+            if !remaining_table_id_set.is_empty() {
+                let notified = self.notify.notified();
+                notified.await;
+            }
+        }
+
+        multi_slice_transform
+    }
+
+    fn set_version(&self, version: u64) {
+        self.version.store(version, AtomicOrdering::SeqCst);
+        self.notify.notify_waiters();
+    }
+
+    fn version(&self) -> u64 {
+        self.version.load(AtomicOrdering::SeqCst)
+    }
+}
+
+#[derive(Default)]
+pub struct SliceTransformManager {
```

Comment:
It would better to add some docs for `SliceTransformManager`

Reply:
ok, i will rename it and add some doc for its background